packages/dds/tree/src/feature-libraries/flex-tree/lazyField.ts

```diff
@@ -69,20 +71,67 @@ import {
 } from "./utilities.js";
 import { UsageError } from "@fluidframework/telemetry-utils/internal";
 
+/**
+ * Reuse fields.
+ * Since field currently own cursors and register themselves for disposal when the node hit end of life,
+ * not reusing them results in memory leaks every time the field is accessed.
+ * Since the fields stay alive until the node is end of life reusing them this way is safe.
+ *
+ * This ins't a perfect solution:
+ *
+ * - This can cause leaks, like map nodes will keep all accessed field objects around. Since other things cause this same leak already, its not too bad.
+ * - This does not cache the root.
+ * - Finding the parent anchor to do the caching on has significant cost.
+ *
+ * Despite these limitations, this cache provides a large performance win in some common cases (over 10x), especially with how simple tree requests far more field objects than necessary currently.
+ */
+const fieldCache: WeakMap<LazyTreeNode, Map<FieldKey, FlexTreeField>> = new WeakMap();
+
 export function makeField(
 	context: Context,
 	schema: FlexFieldSchema,
 	cursor: ITreeSubscriptionCursor,
 ): FlexTreeField {
 	const fieldAnchor = cursor.buildFieldAnchor();
+	let usedAnchor = false;
+
+	const makeThing = (): FlexTreeField => {
```

Comment:
a more descriptive name than "makeThing" would be helpful :)

Reply:
makeFlexField?

Reply:
More like makeFieldFlex 💪 amirite these fields are cached so now 500,000 fields are combined into a single field, that's strooonnggg